### PR TITLE
[POAE7-1437] updated ubuntu build bash script

### PR DESCRIPTION
**Pull request by sijialouintel**
(1) The original web is invalid now.
(2) Thrift with Go should be installed with "go get" command. With "--with-go", go module will only be built and will not be install by later "make install" command.
(3) The golang built in mapd-deps-prebuilt.sh is a old version. If build thrift with go in (2), it will cause version conflict. Besides, mapd-deps-ubuntu.sh also installs go. No need to install it twice with two different versions.

### CODE REVIEW

File: build-omnisci-debug.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
```
Comment (winningsix) [inline build-omnisci-debug.sh:1]: can we comment where this script comes from? Apache license header please.

File: scripts/mapd-deps-prebuilt.sh [hunk 2]
```diff
@@ -103,7 +103,6 @@ if [ "$ID" == "ubuntu" ] ; then
       g++-9 \
       libboost-all-dev \
       libgoogle-glog-dev \
-      golang \
```
Comment (winningsix) [inline scripts/mapd-deps-prebuilt.sh:106]: Let's comment here adding a TODO. And update https://github.com/Intel-bigdata/omniscidb/tree/cider#dependencies to mark this limitation and reason.

File: scripts/omnisci-env.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
```
Comment (winningsix) [inline scripts/omnisci-env.sh:1]: apache license header.
can we update related sections in https://github.com/Intel-bigdata/omniscidb/tree/cider#building

**Review by winningsix — APPROVED** [review]
Overall LGTM. we can create another PR addressing doc issues.

File: build-omnisci-debug.sh (hunk 1 above)
Comment (sijialouintel) [reply build-omnisci-debug.sh:1]: This script is totally a new one, which aims to build omnisci with debug mode.

File: build-omnisci-debug.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
```
Comment (winningsix) [inline build-omnisci-debug.sh:1]: license header starts at 1st line.

File: scripts/mapd-deps-ubuntu.sh [hunk 5]
```diff
@@ -153,8 +153,8 @@ CFLAGS="-fPIC" CXXFLAGS="-fPIC" JAVA_PREFIX=$PREFIX/lib ./configure \
     --with-qt4=no \
     --with-qt5=no \
     --with-java=no \
+    --with-go=no \
```
Comment (winningsix) [inline scripts/mapd-deps-ubuntu.sh:156]: need to unsupport go-lang?

File: scripts/omnisci-env.sh (hunk 3 above)
Comment (winningsix) [inline scripts/omnisci-env.sh:1]: same as above.

**Review by winningsix — COMMENTED** [review]
Overall LGTM a few comments.

File: scripts/mapd-deps-prebuilt.sh (hunk 2 above)
Comment (sijialouintel) [reply scripts/mapd-deps-prebuilt.sh]: Update: no need to change here.
This change has been reverted.

File: scripts/omnisci-env.sh (hunk 3 above)
Comment (sijialouintel) [reply scripts/omnisci-env.sh:1]: added

File: build-omnisci-debug.sh (hunk 4 above)
Comment (sijialouintel) [reply build-omnisci-debug.sh:1]: `#!/usr/bin/env bash` should be put first

File: scripts/mapd-deps-ubuntu.sh (hunk 5 above)
Comment (sijialouintel) [reply scripts/mapd-deps-ubuntu.sh:156]: Even without `--with-go=no`, thrift with go can only be built, which will not be installed by later `make install` command.

File: scripts/omnisci-env.sh (hunk 3 above)
Comment (sijialouintel) [reply scripts/omnisci-env.sh:1]: Apache license header has been added. Will raise another pr to deal with doc related modification.

**Comment by sijialouintel** [comment]
Finished modification. Waiting for review. @Intel-bigdata/bdf-cider-all

File: build-omnisci-debug.sh (hunk 4 above)
Comment (sijialouintel) [reply build-omnisci-debug.sh:1]: reference: https://github.com/openvinotoolkit/openvino/blob/master/scripts/setupvars/setupvars.sh

File: scripts/mapd-deps-ubuntu.sh (hunk 5 above)
Comment (sijialouintel) [reply scripts/mapd-deps-ubuntu.sh:156]: reference: https://thrift.apache.org/lib/go.html

**Pull request merged**